lib/rules/curly.js

```diff
@@ -309,6 +331,115 @@ module.exports = {
                 hasUnsafeIf(statement) && isFollowedByElseKeyword(node);
         }
 
+        /**
+         * Missing curly brace reporting location examples:
+         *
+         * The following location should be reported
+         *
+         * if (foo) bar()
+         *        ^^^
+         *
+         * while(foo) bar();
+         *          ^^^
+         *
+         * do foo() while(bar)
+         * ^^^
+         *
+         * for(;;) bar();
+         *       ^^^
+         *
+         * for (var foo of bar) console.log(foo)
+         *                    ^^^
+         *
+         * for (var foo in bar) console.log(foo)
+         *                    ^^^
+         *
+         * if (foo) {bar()} else bar(1);
+         *                  ^^^^
+         *
+         * Unexpected curly brace reporting location examples:
+         *
+         * The following location should be reported
+         *
+         * if (true) foo(); else { baz(); }
+         *                       ^
+         *
+         * for (;;) { foo(); }
+         *          ^
+         *
+         * while (bar) { foo(); }
+         *             ^
+         *
+         * do{foo();} while(bar);
+         *   ^
+         *
+         * for (var foo of bar) {console.log(foo)}
+         *                      ^
+         *
+         * if (foo) bar() else {bar(1)}
+         *                ^^^^
+         *
+         * Get the location of missing or non missing curly brace.
+         * @param {ASTNode} node the node to check.
+         * @param {string} name name to check.
+         * @param {boolean} isMissing is curly brace missing.
+         * @returns {Object} Position object.
+         */
+        function getReportLoc(node, name, isMissing) {
+            if (name === "else") {
+
+                if (!isMissing) {
+                    const token = sourceCode.getTokenAfter(getElseKeyword(node));
+
+                    return token.loc;
+                }
+
+                return getElseKeyword(node).loc;
+            }
+
+            if (isMissing) {
+
+                if (name === "for" || name === "do") {
+                    const token = sourceCode.getTokenBefore(node.body);
+
+                    return {
+                        start: token.loc.end,
+                        end: node.body.loc.start
+                    };
+                }
+
+                if (name === "for-of" || name === "for-in") {
+                    return {
+                        start: node.right.loc.end,
+                        end: node.body.loc.start
+                    };
+                }
```

Comment:
Why should be `for-of`, `for-in`, `if` and `while` different than `for` and `do`:

![image](https://user-images.githubusercontent.com/44349756/88417009-799eb000-cde1-11ea-8529-8a83ccb6f9f6.png)

Reply:
I will check this.
It might have come along with https://github.com/eslint/eslint/pull/13282/commits/b3330be4930a8c4d2f9d65b599d73d6e94cf9ba2

Reply:
done